ruby/sdk-phase-1.md

```diff
@@ -0,0 +1,439 @@
+# Ruby - Phase 1
+
+The Ruby SDK will be implemented in two loosely defined phases:
+
+* Phase 1 - Initial project, activity worker, client, and test framework
+* Phase 2 - Workflow worker and replayer
+
+This proposal is for phase 1. While proposals and SDK may be done in large phases, work is actually incremental.
+
+Notes for this proposal:
+
+* Everything is subject to change both during proposal time and afterwards during implementation.
+* For each place a decision is made, a "💭 Why X? ..." aside is present explaining why.
+* For each place an open question exists, a "❓ X?" exists to ask it.
+
+## Overview
+
+The Ruby SDK will be backed by Rust [sdk-core](https://github.com/temporalio/sdk-core) and will follow the model set by
+other SDKs.
+
+Much of the work for this proposal and the implementation comes the past proposals in [previous](previous)
+and the implementation built from that. 💭 Why start a new proposal/implementation instead of adapting that? The
+constraints, approach, and requirements changed enough to justify a clean slate and while many things may be leveraged,
+it never went beyond alpha anyways.
+
+Lots of approaches taken here are similar to recent Python and .NET SDKs.
+
+### Requirements
+
+These are the general requirements of the project
+
+* Ruby >= 3.1
```

Comment:
Wondering what we've done to validate this is acceptable.

Reply:
All older versions of Ruby are EOL; I don't think we need to support EOL software: https://www.ruby-lang.org/en/downloads/

Reply:
Correct: https://en.wikipedia.org/wiki/History_of_Ruby. But all of this is still pending user feedback and so we'll likely be surveying the community on how acceptable they consider this.